analyses/sample-distribution-analysis/01-filter-across-types.R

```diff
@@ -0,0 +1,146 @@
+# This script filters the given dataset to produce a summarized visualization
+# of key variables within the dataset.
+#
+# Chante Bethell for CCDL 2019
+
+
+# Load/install packages. This will be removed once a Dockerfile is established.
+source("00-install-packages.R")
+
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Function to filter based on primary_site 
+location_fn <- function(location) {
+  disease_type_vector <- brain_location %>%
+    dplyr::arrange(dplyr::desc(n)) %>%
+    dplyr::filter(stringr::str_detect(primary_site, location)) %>%
+    dplyr::pull(disease_type_new)
+  unique(disease_type_vector)
+}
+
+# Create directories to hold the output.
+if (!dir.exists("results")) {
+  dir.create("results")
+}
+if (!dir.exists("plots")) {
+  dir.create("plots")
+}
+
+# Read in dataset
+df2 <- data.frame(readr::read_tsv(
+  file.path("..", "..", "data", "pbta-histologies.tsv")
+))
+
+# Remove na's
+df2 <- df2 %>%
+  dplyr::filter(!is.na(disease_type_new))
+
+# data.frame with the count of each unique cancer type expression
+disease_expression <- df2 %>%
+  dplyr::group_by(disease_type_new) %>%
+  dplyr::count(name = "count") %>%
+  dplyr::arrange(dplyr::desc(count))
+
+# Calculate the total count of the dataset
+sum_count <- sum(disease_expression$count)
+
+# Create a percent variable
+disease_expression <- disease_expression %>%
+  dplyr::mutate(percent = paste0(((count / sum_count) * 100), "%"))
```

Comment:
Now that you've pushed the changes. I wanted to note that the numbers look ever so slightly different to make sure you are aware of that and I think the last thing that function was doing was performing a rounding step, which can likely be accomplished with something like:
```suggestion
  dplyr::mutate(percent = paste0((round(count / sum_count, 2) * 100), "%"))
```

Reply:
Ah, very true. I will make that change now and check the numbers before committing the change.